### PR TITLE
Showroundstatsfix and pointer video

### PR DESCRIPTION
**Pull request by kuhlkrein**

### CODE REVIEW

**Comment by kuhlkrein** [comment]
How can I remove the egg and eggGame files from this commit ?

**Comment by Tohaker** [comment]
You'll need to pull the latest master and rebase your branch

**Comment by kuhlkrein** [comment]
@YannRobert , @Tohaker I think it is ok now right ? sorry it was a bit long..

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java [hunk 1]
```diff
@@ -96,6 +102,29 @@ protected GameContext(
...
+                pointer.setCenterX(((MouseEvent) e).getX());
+                pointer.setCenterY(((MouseEvent) e).getY());
+            } else if (e.getEventType() == GazeEvent.GAZE_MOVED) {
+                pointer.setCenterX(((GazeEvent) e).getX());
+                pointer.setCenterY(((GazeEvent) e).getY());
+            }
+            pointer.toFront();
+        };
+        root.addEventFilter(MouseEvent.ANY, pointerEvent);
+        root.addEventFilter(GazeEvent.ANY, pointerEvent);
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: Something like this would be better :

```
root.addEventFilter(GazeEvent.GAZE_MOVED, e -> {
                pointer.setCenterX(e.getX());
                pointer.setCenterY(e.getY());
});
```

Same for the MouseEvent

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java [hunk 2]
```diff
@@ -96,6 +102,29 @@ protected GameContext(
 
         this.gamePanelDimensionProvider = new GamePanelDimensionProvider(() -> root, gazePlay::getPrimaryScene);
         this.randomPositionGenerator = new RandomPanePositionGenerator(gamePanelDimensionProvider);
+
+        if (this.getConfiguration().isVideoRecordingEnabled()) {
+            this.createPointer();
+        }
+    }
+
+    public void createPointer() {
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: When I read createPointer() I ancipate the method would just create (and return) a Pointer, nothing else, no side effect.

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java [hunk 3]
```diff
@@ -76,6 +79,9 @@ public static void updateConfigPane(final Pane configPane, Stage primaryStage) {
 
     private final Pane gamingRoot;
 
+    private EventHandler<Event> pointerEvent;
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: this is probably not needed as a field.

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java [hunk 4]
```diff
@@ -76,6 +79,9 @@ public static void updateConfigPane(final Pane configPane, Stage primaryStage) {
...
+    private Circle pointer;
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: this is probably not needed as a field

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/stats/AreaOfInterest.java [hunk 5]
```diff
@@ -553,6 +553,7 @@ private void playButtonPressed() {
                 // player.setRate(1);
                 player.stop();
                 player.play();
+                // this.stats.startVideoRecording();
```
Comment (YannRobert) [inline gazeplay/src/main/java/net/gazeplay/ui/scenes/stats/AreaOfInterest.java]: why is that needed ?

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java (hunk 1 above)
Comment (YannRobert) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: mouse and gaze can move in different directions/positions

so either
- we need a pointer for both of them
- we only need a pointer for gaze, as the mouse already has a native pointer

**Comment by kuhlkrein** [comment]
@YannRobert I need to keep pointer and pointereEvent as a field now that I'm deleting them when exiting a game

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java (hunk 1 above)
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: @YannRobert , the point is that some eye-trackers are making the mouse move too, but yes, in this cas we can have two pointers, maybe it's the best idea

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/stats/AreaOfInterest.java (hunk 5 above)
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/stats/AreaOfInterest.java]: Sry I forgot to remove the comment, I think the idea was to record a video on wich the AOI appear, but it is clearly not working, and it's not a good way to do it. I deleted the comment, thank you

File: gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java (hunk 2 above)
Comment (kuhlkrein) [reply gazeplay/src/main/java/net/gazeplay/ui/scenes/ingame/GameContext.java]: I changed the name with pointersInit(), so we're not waiting for any return value and we understand that there are side effetcs
